src/api/types.js

```diff
@@ -17,3 +18,29 @@
  * @import { Request } from '@hapi/hapi'
  * @import { Db } from 'mongodb'
  */
+
+/**
+ * @template T
+ * @typedef {object} Result
+ * @property {T[]} data - The array of data items.
+ * @property {Meta} meta - The metadata about the result.
+ */
+
+/**
+ * @typedef {object} Meta
+ * @property {PaginationResult} [pagination] - The pagination details.
+ */
+
+/**
+ * @typedef {object} PaginationResult
+ * @property {number} page - The current page number.
+ * @property {number} perPage - The number of items per page.
+ * @property {number} totalItems - The total number of items available.
+ * @property {number} totalPages - The total number of pages available.
+ */
+
+/**
+ * @typedef {object} PaginationOptions
+ * @property {number} [page] - The current page number.
+ * @property {number} [perPage] - The number of items per page.
+ */
```

Comment:
Would it help if we just picked the properties we needed from `PaginationResult`?

Then can use `Partial<PaginationOptions>` when they're optional

```suggestion
/**
 * @typedef {Pick<PaginationResult, 'page' | 'perPage'>} PaginationOptions
 */
```

Reply:
Yeah that's def more concise tbf.